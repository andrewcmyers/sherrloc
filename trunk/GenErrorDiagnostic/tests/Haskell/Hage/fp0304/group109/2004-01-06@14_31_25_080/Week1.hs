module Week1 where

voegIn :: Int -> [Int] -> [Int]
voegIn a [] = [a]
voegIn a (x:xs) | a<= x = a:x:xs
                | a> x = x: (voegIn a xs)

insertionSort:: [Int] -> [Int]
insertionSort [] = []
insertionSort (x:xs) = voegIn x (insertionSort xs)




























vanafLaatste :: [Int] -> [Int]
vanafLaatste (x:xs) | x /= 0 = vanafLaatste xs
                    | x == 0 && (laatste0 x) = xs
                    | x == 0 && not (laatste0 x) = vanafLaatste xs

laatste0 :: [Int] -> Bool
laatste0 [] = True
laatste0 (x:xs) | x /= 0 = laatste0 xs
                | x == 0 = False



alles [] a b = []
alles (x:xs) a b | x /= 0 = a
                 | x == 0 = b


totEerste (x:xs) = alles (x:xs) (x:(totEerste xs)) []


-- x should be xs
-- 41,43-43   42,47-47
