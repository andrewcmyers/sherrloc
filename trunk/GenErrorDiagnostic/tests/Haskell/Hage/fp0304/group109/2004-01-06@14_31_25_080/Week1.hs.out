Compiling Week1.hs
(41,34)-(41,42): Type error in application
 expression       : laatste0 x
 term             : laatste0
   type           : [Int] -> Bool
   does not match : Int   -> Bool

(42,38)-(42,46): Type error in application
 expression       : laatste0 x
 term             : laatste0
   type           : [Int] -> Bool
   does not match : Int   -> Bool

Compilation failed with 2 errors
